scripts/bench-folder.ts

```diff
@@ -1,147 +1,164 @@
-
-  // We create a few helper scripts to make the benchmarking easier
-  xfs.writeFileSync(ppath.join(binaryDir, `yarn-repo`), [
-    `#!/bin/bash\n`,
-    `cd ${npath.fromPortablePath(repoDir)}\n`,
-    `unset YARN_IGNORE_PATH\n`,
-    `export PATH="${process.env.PATH}"\n`,
-    `exec yarn $@\n`,
-  ].join(``));
-
-  xfs.writeFileSync(ppath.join(binaryDir, `bench-commit`), [
-    `#!/bin/bash\n`,
-    `cd ${npath.fromPortablePath(temp)}\n`,
-    `git add . && (git diff-index --quiet HEAD || git commit -m Commit > /dev/null)\n`,
-  ].join(``));
-
-  xfs.writeFileSync(ppath.join(binaryDir, `bench-reset`), [
-    `#!/bin/bash\n`,
-    `cd ${npath.fromPortablePath(temp)}\n`,
-    `git reset --hard HEAD && git clean -fdx\n`,
-  ].join(``));
-
-  xfs.writeFileSync(ppath.join(binaryDir, `bench-run`), [
-    `#!/bin/bash\n`,
-    `cd ${npath.fromPortablePath(temp)}\n`,
-    `\n`,
-    `PATH="$(pwd)/bin/hyperfine:$PATH" hyperfine --warmup 1 \\\n`,
-    `  --export-markdown=.git/yarn-bench \\\n`,
-    `  --prepare 'bench-reset && bash "${npath.fromPortablePath(temp)}"/bench-prepare.sh' \\\n`,
-    `  'before' \\\n`,
-    `  'after'\n`,
-    `\n`,
-    // macOS
-    `if which pbcopy >/dev/null 2>&1; then\n`,
-    `  pbcopy < .git/yarn-bench\n`,
-    // Linux
-    `elif which xclip >/dev/null 2>&1; then\n`,
-    `  xclip -selection clipboard < .git/yarn-bench\n`,
-    `fi\n`,
-  ].join(``));
-
-  // We don't want Yarn to be called directly, since it'd be a global copy we don't control
-  xfs.writeFileSync(ppath.join(binaryDir, `yarn`), [
-    `#!/bin/bash\n`,
-    `echo "Don't run Yarn directly:"\n`,
-    `echo\n`,
-    `echo "  - if the output is the same between both versions, use yarn-before or yarn-after instead"\n`,
-    `echo "  - otherwise, add the call to the bench-prepare.sh script"\n`,
-    `exit 1\n`,
-  ].join(``));
-
-  // We also create another binary dir, just to clean up the command line we show in Hyperfine
-  xfs.mkdirSync(ppath.join(binaryDir, `hyperfine`));
-
-  xfs.writeFileSync(ppath.join(binaryDir, `hyperfine/before`), [
-    `#!/bin/bash\n`,
-    `cd ${npath.fromPortablePath(temp)}\n`,
-    `PATH="$(pwd)/bin/before:$PATH" bash bench-script.sh\n`,
-  ].join(``));
-
-  xfs.writeFileSync(ppath.join(binaryDir, `hyperfine/after`), [
-    `#!/bin/bash\n`,
-    `cd ${npath.fromPortablePath(temp)}\n`,
-    `PATH="$(pwd)/bin/after:$PATH" bash bench-script.sh\n`,
-  ].join(``));
-
-  // Those two scripts are meant to be written by the user (but not run directly, so not executable and no shebang)
-  xfs.writeFileSync(ppath.join(temp, `bench-prepare.sh`), ``);
-  xfs.writeFileSync(ppath.join(temp, `bench-script.sh`), `yarn install\n`);
-
-  // General Yarn configuration
-  xfs.writeJsonSync(ppath.join(temp, Filename.rc), {
-    globalFolder: `.yarn/global`,
-  });
-
-  xfs.writeJsonSync(ppath.join(temp, Filename.manifest), {
-    name: `benchmark`,
-    private: true,
+runExit(class extends Command {
```

Comment:
Added two features here to make benchmarking easier: `bench-import <name>` will copy the benchmark manifest from one of the presets, and `--branch` can be used to test the current build against a different branch or commit than just `master`.